CartoDB/carto-react
# Comparative Category Widget

# Description

This PR implements part of the work proposed in RFC https://github.com/CartoDB/carto-react/issues/479

This widget uses the component `AnimatedNumber` introduced in PR #504. Because of that, it includes the changes introduced in PR #504. Because of that, this PR should be merged only after PR #504 is merged

Changes added in this PR include:
* A new widget called `ComparativeCategoryWidgetUI` in the `widgets` folder
* Typescript typings for `ComparativeCategoryWidgetUI` and the exported enum `ORDER_TYPES`
* JSDoc typings comments for `ComparativeCategoryWidgetUI` and the exported enum `ORDER_TYPES`
* Prop-types validation for `ComparativeCategoryWidgetUI` and all the subcomponents included
* A basic storybook story for `ComparativeComparativeWidgetUI` where all prop combinations can be tested and tweaked. A screenshot of this storybook is added as visual feedback.

![Captura de pantalla 2022-10-25 a las 12 58 40](https://user-images.githubusercontent.com/6652814/197756015-7b787de4-b657-4bd8-8fc4-595156b8b564.png)

## Type of change

- Feature

# Acceptance

How to validate the feature:

1. Open the local storybook running `yarn storybook:start` in the `react-ui` folder.
2. Pick a set of properties to use for testing by tweaking the storybook configuration.
3. Use this set of properties to render a `ComparativeCategoryWidgetUI` in your sample project, by importing `ComparativeCategoryWidgetUI` from `@carto/react-ui`. You can use [this branch](https://github.com/CartoDB/carto-react-template/tree/feature/comparative-widgets) of `carto-react-template` to skip this step if you want, as a test screen is already setup there. Bear in mind that if this feature is not published to npm you will have to link your local version of `@carto/react-*` packages with `yarn link`.
4. Check that the rendered version of the widget in your sample project looks the same as it looks in the storybook.

# Basic checklist

- Good PR name
- Shortcut link
- Changelog entry
- Just one issue per PR
- GitHub labels
- Proper status & reviewers
- Tests
- Documentation
